endstate_correction/smc.py

```diff
@@ -0,0 +1,197 @@
+# Perform SMC sampling for endstate correction
+
+import logging
+from typing import List
+
+import numpy as np
+from mdtraj import Trajectory
+from openmm.app import Simulation
+from scipy.special import logsumexp
+from tqdm import tqdm
+from itertools import accumulate
+from endstate_correction.constant import kBT, temperature
+
+logger = logging.getLogger(__name__)
+
+
+class Resampler:
+    def __init__(self):
+        pass
+
+    def stratified_resampling(self, samples: List, weights: List[float]) -> List:
```

Comment:
Thanks, would be good to test this with some simple unit tests. E.g. repeatedly resampling samples with weights `0.5` and `0.5` should give you the same two samples.